test/middleware/retry_test.rb

```diff
@@ -212,5 +229,42 @@ def test_should_rewind_files_on_retry
       assert_equal 3, @times_called
       assert_equal 2, rewound
     end
+
+    def test_should_retry_retriable_response
+      assert_raises(Faraday::Error::RetriableResponse) {
+        params = { status: 429 }
+        conn(:max => 1, :retry_statuses => 429).get("/throttled", params)
+      }
+      assert_equal 2, @times_called
+    end
+
+    def test_should_not_retry_non_retriable_response
+      params = { status: 503 }
+      conn(:max => 1, :retry_statuses => 429).get("/throttled", params)
+
+      assert_equal 1, @times_called
+    end
+
+    def test_interval_if_retry_after_present
+      started = Time.now
+
+      assert_raises(Faraday::Error::RetriableResponse) {
+        params = { :retry_after => 0.5 }
+        conn(:max => 1, :interval => 0.1, :retry_statuses => [429]).get("/throttled", params)
+      }
+
+      assert_in_delta 0.5, Time.now - started, 0.04
```

Comment:
I'm a bit worried by this assertion, it might create false-positives if test execution is too slow.
Any reason why we don't simply test that `Time.now - started > 0.5`?

Reply:
Agree. We don't care about delta in this case.